game/src/contrib/systems/CollisionSystem.java

```diff
@@ -19,18 +20,48 @@ public CollisionSystem() {
         super(CollideComponent.class);
     }
 
+    /**
+     * checks every CollideEntity with each other CollideEntity and checks them for a collision.
+     *
+     * <p>Some improvements are done so there is only one collision check between two Entities based
+     * on their IDs. When Entity A does collide with Entity B it also means B collides with A.
+     */
     @Override
     public void execute() {
-        getEntityStream()
-                .flatMap(
-                        a ->
-                                getEntityStream()
-                                        .filter(b -> a.id() < b.id())
-                                        .map(b -> buildData(a, b)))
-                .forEach(this::onEnterLeaveCheck);
+        getEntityStream().flatMap(this::createDataPairs).forEach(this::onEnterLeaveCheck);
+    }
+
+    /**
+     * Creates a Stream where every other Entity has a higher id.
+     *
+     * @param a Entity which is the lower id partner
+     * @return the stream which contains every valid pair of Entities
+     */
+    private Stream<CollisionData> createDataPairs(Entity a) {
+        return getEntityStream().filter(b -> isIDSmallerThen(a, b)).map(b -> newDataPair(a, b));
+    }
+
+    /**
+     * Makes sure only Entities with a higher id are paired up.
+     *
+     * <p>This Prevents a pair of Entities to be checked double or with itself.
+     *
+     * @param a first Entity
+     * @param b second Entity
+     * @return true when the id of b is bigger then the one from a
+     */
+    private static boolean isIDSmallerThen(Entity a, Entity b) {
```

Comment:
ich frage mich grad, warum das (a) static ist und (b) im system? ist das nicht eine funktionalität, die auf daten der entity ausgeführt wird, also besser dorthin gehört?

Reply:
static war ein kleiner Fehler, aber b liegt daran, dass es kein Standard Compare ist. Default ist ja die Differenz zwischen den Entitäten. Dies hier ist ein true, false. Könnte aber eine neue Funktionalität von Entity werden.

Reply:
Hmmm, für mich klingt es immer noch nach Feature-Neid :) ... Die Methode operiert eigentlich nur auf den (inneren) Daten der übergebenen Parameter. Ich glaube, das wäre besser in `Entity` angesiedelt. Und wenn Du schon dabei bist, könnte man überlegen, ob man das im Rahmen von `Comparable` und Co macht?

Reply:
Habe dafür mal lieber #754 erstellt. Da dies glaube, ich hier aufhält und ggfls. noch mal durchdacht werden muss, was alles mit eingeschlossen sein soll.